src/Nancy/HttpLink.cs

```diff
@@ -0,0 +1,283 @@
+        {
+            get { return this.GetParameterValue<HttpLinkRelation>("rel"); }
+        }
+
+        /// <summary>
+        /// Gets the target URI.
+        /// </summary>
+        public Uri TargetUri
+        {
+            get { return this.targetUri; }
+        }
+
+        /// <summary>
+        /// The optional title parameter is used to label the destination of a link such that it can be used as a human-readable identifier (e.g., a menu entry) in the language indicated by the HTTP Content-Language header (if present).
+        /// </summary>
+        public string Title
+        {
+            get { return this.GetParameterValue<string>("title"); }
+        }
+
+        /// <summary>
+        /// The optional type parameter is a hint indicating what the media type of the result of dereferencing the link should be. Note that this is only a hint; for example, it does not override the HTTP Content-Type header of a HTTP response obtained by actually following the link.
+        /// </summary>
+        public MediaRange Type
+        {
+            get { return this.GetParameterValue<MediaRange>("type"); }
+        }
+
+        /// <summary>
+        /// Indicates whether the current object is equal to another object of the same type.
+        /// </summary>
+        /// <param name="other">An object to compare with this object.</param>
+        /// <returns>
+        /// true if the current object is equal to the <paramref name="other" /> parameter; otherwise, false.
+        /// </returns>
+        public bool Equals(HttpLink other)
+        {
+            if (ReferenceEquals(other, null))
+            {
+                return false;
+            }
+
+            if (this.TargetUri != other.TargetUri)
+            {
+                return false;
+            }
+
+            foreach (var parameter in other.Parameters)
+            {
+                object parameterValue;
+                if (!this.parameters.TryGetValue(parameter.Key, out parameterValue))
+                {
+                    return false;
+                }
+
+                if (parameterValue == null || parameter.Value == null)
+                {
+                    continue;
+                }
+
+                if (!parameterValue.Equals(parameter.Value))
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Determines whether the specified <see cref="System.Object" />, is equal to this instance.
+        /// </summary>
+        /// <param name="other">The <see cref="System.Object" /> to compare with this instance.</param>
+        /// <returns>
+        ///   <c>true</c> if the specified <see cref="System.Object" /> is equal to this instance; otherwise, <c>false</c>.
+        /// </returns>
+        public override bool Equals(object other)
+        {
+            return ((IEquatable<HttpLink>)this).Equals(other as HttpLink);

```

Comment:
Now that the method isn't implemented explicitly anymore, we can remove this horrible cast :wink: